monai/transforms/lazy/functional.py

```diff
@@ -0,0 +1,201 @@
+    matrix_factory = MatrixFactory(dim_count, get_backend_from_tensor_like(data), get_device_from_tensor_like(data))
+
+    # set up the identity matrix and metadata
+    cumulative_matrix, cumulative_extents = starting_matrix_and_extents(matrix_factory, data)
+
+    # set the various resampling parameters to an initial state
+    cur_mode = None
+    cur_padding_mode = None
+    cur_device = None
+    cur_dtype = None
+    cur_shape = data.shape
+
+    for meta_matrix in pending_:
+        next_matrix = meta_matrix.matrix
+        # print("intermediate matrix\n", matrix_from_matrix_container(cumulative_matrix))
+        cumulative_matrix = matmul(cumulative_matrix, next_matrix)
+        cumulative_extents = [matmul(e, cumulative_matrix) for e in cumulative_extents]
+
+        new_mode = meta_matrix.metadata.get("mode", None)
+        new_padding_mode = meta_matrix.metadata.get("padding_mode", None)
+        new_device = meta_matrix.metadata.get("device", None)
+        new_dtype = meta_matrix.metadata.get("dtype", None)
+        new_shape = meta_matrix.metadata.get("shape_override", None)
+
+        mode_compat = metadata_is_compatible(cur_mode, new_mode)
+        padding_mode_compat = metadata_is_compatible(cur_padding_mode, new_padding_mode)
+        device_compat = metadata_is_compatible(cur_device, new_device)
+        dtype_compat = metadata_dtype_is_compatible(cur_dtype, new_dtype)
+
+        if mode_compat is False or padding_mode_compat is False or device_compat is False or dtype_compat is False:
+            # carry out an intermediate resample here due to incompatibility between arguments
+            kwargs = prepare_args_dict_for_apply(cur_mode, cur_padding_mode, cur_device, cur_dtype)
+
+            cumulative_matrix_ = matrix_from_matrix_container(cumulative_matrix)
+            a = Affine(norm_coords=False, affine=cumulative_matrix_, **kwargs)
+            data, _ = a(img=data)
+
+            cumulative_matrix, cumulative_extents = starting_matrix_and_extents(matrix_factory, data)
+
+        cur_mode = cur_mode if new_mode is None else new_mode
+        cur_padding_mode = cur_padding_mode if new_padding_mode is None else new_padding_mode
+        cur_device = cur_device if new_device is None else new_device
+        cur_dtype = cur_dtype if new_dtype is None else new_dtype
+        cur_shape = cur_shape if new_shape is None else new_shape
+
+    kwargs = prepare_args_dict_for_apply(cur_mode, cur_padding_mode, cur_device, cur_dtype)
+
+    cumulative_matrix_ = matrix_from_matrix_container(cumulative_matrix)
+
+    # print(f"applying with cumulative matrix\n {cumulative_matrix_}")
+    a = Affine(norm_coords=False, affine=cumulative_matrix_, spatial_size=cur_shape[1:], normalized=False, **kwargs)
+    data, tx = a(img=data)
+    if isinstance(data, MetaTensor):
+        data.clear_pending_transforms()
+        for p in pending_:
+            data.affine = p.matrix.data
+            data.push_applied_operation(p)
```

Comment:
thanks for the updates, could you create an array transform example based on this? the previous #4922's transform seems to be out of date (https://github.com/Project-MONAI/MONAI/blob/62f8172f020c3c8b6ac182b6b4ca5c3805f9877b/monai/transforms/atmostonce/array.py)